src/jit/gentree.cpp

```diff
@@ -10694,7 +10694,15 @@ void Compiler::gtDispConst(GenTree* tree)
         case GT_CNS_INT:
             if (tree->IsIconHandle(GTF_ICON_STR_HDL))
             {
-                printf(" 0x%X \"%S\"", dspPtr(tree->gtIntCon.gtIconVal), eeGetCPString(tree->gtIntCon.gtIconVal));
+                const wchar_t* str = eeGetCPString(tree->gtIntCon.gtIconVal);
+                if (str)
```

Comment:
No you should leave the output the same as before and if you aren't fixing the underlying issue just printout a not-implemented message.

Like this:

 ```
const wchar_t* str = eeGetCPString(tree->gtIntCon.gtIconVal);
 if (str != nullptr)
 {
  printf(" 0x%X \"%S\"", dspPtr(tree->gtIntCon.gtIconVal), eeGetCPString(tree->gtIntCon.gtIconVal));
}
else
{
   // Note that eeGetCPString isnt implemented on Linuxand instead r\always returns nullptr
   pintf( eeGetCPString( not implemented for target platform");
}
```